### PR TITLE
[ML] Incremental binomial logistic loss

### PR DESCRIPTION
**Pull request by tveasey**
This implements the incremental version of the binomial logistic loss. We add on a penalty which is the cross-entropy of the old and new tree predictions on the original training data.

### CODE REVIEW

File: lib/maths/CBoostedTreeLoss.cc [hunk 1]
```diff
@@ -4,6 +4,7 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
+#include "maths/CBoostedTreeUtils.h"
```
Comment (droberts195) [inline lib/maths/CBoostedTreeLoss.cc:7]:
```suggestion
#include <maths/CBoostedTreeUtils.h>
```

**Review by droberts195 — APPROVED** [review]
LGTM - just one nit about using angle brackets for library includes

File: lib/maths/CBoostedTreeLoss.cc (hunk 1 above)
Comment (tveasey) [reply lib/maths/CBoostedTreeLoss.cc]: Done in [c83ffe6](https://github.com/elastic/ml-cpp/pull/1901/commits/c83ffe6f36344efce60c1b32d68f0cb671b2e48a).
